tests/Authorization/ProviderTest.php

```diff
@@ -164,12 +194,143 @@ public function testProvideNoErrCodeException()
         $client = new GuzzleHttp\Client(['handler' => $stack,]);
 
         $provider = new Ubki\Authorization\Provider(
-            new Ubki\Config('test-provider-username', 'test-provide-password'),
             $client,
             $this->logger
         );
 
         /** @noinspection GuzzleHttp\Exception\ClientException */
-        $response = $provider->provide();
+        $response = $provider->provide($this->config);
+    }
+
+    /**
+     * @expectedException Wearesho\Bobra\Ubki\Authorization\UnsupportedModeException
+     * @throws GuzzleHttp\Exception\GuzzleException
+     * @throws Ubki\Authorization\Exception
+     */
+    public function testProvideInvalidMode(): void
+    {
+        $response = '<?xml version="1.0" encoding="UTF-8" standalone="yes"?><doc><auth/></doc>'; // phpcs:ignore
+        $container = [];
+        $history = GuzzleHttp\Middleware::history($container);
+        $mock = new GuzzleHttp\Handler\MockHandler([
+            new GuzzleHttp\Psr7\Response(400, [], $response),
+        ]);
+        $stack = GuzzleHttp\HandlerStack::create($mock);
+        $stack->push($history);
+
+        $client = new GuzzleHttp\Client(['handler' => $stack,]);
```

Comment:
Do we need such mocked guzzle client in this test case? Will mock handler and container be used?